### PR TITLE
Grafana packages: Remove E2E workspace

### PR DESCRIPTION
**Pull request by sunker**
**What is this feature?**

In a previous [PR](https://github.com/grafana/grafana/pull/85619) targeting G11, a deprecation notice was added to the e2e workspace readme. This PR removes the e2e workspace entirely and any code referencing the `@grafana/e2e` package. 

As soon as `grafana/e2e@11.0.0` is out the door, it will be deprecated in npm. 

**Why do we need this feature?**

`@grafana/e2e` is deprecated and superseded by [`@grafana/plugin-e2e`](https://www.npmjs.com/package/@grafana/plugin-e2e?activeTab=readme).  

**Who is this feature for?**

Grafana core contributors. 

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana/issues/85625 and https://github.com/grafana/grafana/issues/77484

**Special notes for your reviewer:**

I wonder if `@grafana/e2e` is being referenced elsewhere? Grapbl? Release process? cc @grafana/grafana-release-guild

### CODE REVIEW

File: pkg/infra/httpclient/httpclientprovider/log_request.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+)
+
+func LogRequestMiddleware(logger log.Logger) httpclient.Middleware {
+	return httpclient.NamedMiddlewareFunc("log-request", func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
+		return httpclient.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			logger.Info("============ outgoing request", "URL", req.URL)
+			return next.RoundTrip(req)
+		})
+	})
+}
```
Comment (ashharrison90) [inline pkg/infra/httpclient/httpclientprovider/log_request.go:17]: this seems out of place 😅

**Review by ashharrison90 — COMMENTED** [review]
amazing 👏 

just noticed 1 weird thing, otherwise looks great 🥳

File: pkg/infra/httpclient/httpclientprovider/log_request.go (hunk 1 above)
Comment (sunker) [reply pkg/infra/httpclient/httpclientprovider/log_request.go]: oh - thanks for finding this! 😅

**Review by jackw — APPROVED** [review]
Changes LGTM! 🚀 

> I wonder if @grafana/e2e is being referenced elsewhere? Grapbl? Release process?

drone-pipeline runs grabpl [here](https://github.com/grafana/grafana-drone-extension/blob/main/pluginjson/pipeline-ci.go#L251) which in turn calls grafana-e2e [here](https://github.com/grafana/build-pipeline/blob/main/pkg/subcmd/plugin/e2etests.go#L91). This is related to plugin dependencies so I don't think this PR would cause issues there. Grafana itself was decoupled from this package a while back. Certainly worth having a second approval from release-guild though.

**Pull request merged**